src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java

```diff
@@ -1148,6 +1162,37 @@ void processCancelRequest() {
     assertPatronHoldFieldsAreNull((TransactionPatronHold) transactionAfter.getHold());
   }
 
+  @ParameterizedTest
+  @EnumSource(names = {"ITEM_RECEIVED"})
+  @Sql(scripts = {
+          "classpath:db/central-server/pre-populate-central-server.sql",
+          "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql"
+  })
+  void testFinalCheckInWithDeleteVirtualRecord(InnReachTransaction.TransactionState state) {
+    var transactionHoldDTO = createTransactionHoldDTO();
+    var transactionBefore = fetchPrePopulatedTransaction();
+    var configurationDto =
+            deserializeFromJsonFile("/configuration/configuration-details-example.json", ConfigurationDTO.class);
+    when(configurationService.fetchConfigurationsDetailsByModule(any())).
+            thenReturn(ResultList.asSinglePage(configurationDto));
+
+    transactionBefore.setState(state);
+    repository.save(transactionBefore);
+
+    var responseEntity = testRestTemplate.exchange(
+            "/inn-reach/d2ir/circ/finalcheckin/{trackingId}/{centralCode}", HttpMethod.PUT,
+            new HttpEntity<>(transactionHoldDTO, headers), InnReachResponseDTO.class,
+            PRE_POPULATED_TRACKING1_ID, PRE_POPULATED_CENTRAL_CODE);
+
+    var transactionAfter = fetchPrePopulatedTransaction();
```

Comment:
Shouldn't this call to check the data go inside of the `await().untilAsserted` because you want it to be checked after the default delay? I think the default delay is 10 seconds. If you fetch the data before it might work, but it also might fail because the scheduled operation hasn't completed yet.

Reply:
var transactionAfter = fetchPrePopulatedTransaction();
I have changed the call inside the await() and set for default delay with atMost.

Reply:
Ok thanks Nirmal. I think we can call this done. From what I understand this will be merged into the 318 branch and then that will be what gets merged into master. I'll leave it to you or to @gurleenkaurbp to handle that as you like.

Reply:
right @steveellis this will be merged into the 318 branch